test/validation/api/timer/timer.c

```diff
@@ -1218,8 +1282,27 @@ static int worker_entrypoint(void *arg ODP_UNUSED)
 	odp_pool_t tbp = global_mem->tbp;
 	uint32_t num_timers = global_mem->timers_per_thread;
 	uint64_t min_tmo = global_mem->param.min_tmo;
+	odp_queue_param_t queue_param;
+	odp_queue_type_t queue_type = ODP_QUEUE_TYPE_PLAIN;
+	odp_thrmask_t thr_mask;
+	struct thread_args *thr_args = arg;
+	uint64_t sched_tmo;
 
-	queue = odp_queue_create("timer_queue", NULL);
+	odp_queue_param_init(&queue_param);
+	if (thr_args->queue_type == ODP_QUEUE_TYPE_PLAIN) {
+		queue_param.type = ODP_QUEUE_TYPE_PLAIN;
+		queue_type = ODP_QUEUE_TYPE_PLAIN;
+	} else {
+		odp_thrmask_zero(&thr_mask);
+		odp_thrmask_set(&thr_mask, odp_thread_id());
+		queue_param.type = ODP_QUEUE_TYPE_SCHED;
+		queue_param.sched.sync = ODP_SCHED_SYNC_PARALLEL;
+		queue_type = ODP_QUEUE_TYPE_SCHED;
+		queue_param.sched.group = odp_schedule_group_create(NULL,
```

Comment:
Please check the return value here. There was a bug in the basic scheduler implementation, which caused `odp_schedule_group_create()` to fail when trying to create a large number of groups.

This PR needs to be rebased once Petri has created a fix commit.

Reply:
The subject line of the validation test commit should be `validation: timer: check timer capability`

Reply:
The caller function limits the number of threads to number of available schedule groups. But it makes sense to error check here just to be safe and prevent ub.
I will send the updated version, fixing the commit title too.

Thanks for reviewing

Reply:
Here's PR for the fix
https://github.com/OpenDataPlane/odp/pull/1144

Reply:
I will wait for the fix to be merged and then rebase my PR. Thanks!

Reply:
Fix is now merged. Add my review-by tag to the next version. If it passes tests, it would be then ready to be merged.

Reply:
Pushed new version with reviewed-by tag.
Thanks.